r/NEWS.md

```diff
@@ -55,6 +55,7 @@ See `vignette("install", package = "arrow")` for details.
 * Factor types are now preserved in round trip through Parquet format (#6135, @yutannihilation)
 * Reading an Arrow dictionary type coerces dictionary values to `character` (as R `factor` levels are required to be) instead of raising an error
 * Many improvements to Parquet function documentation (@karldw, @khughitt)
+* Can now infer the type of an R `list` and create a ListArray when all list elements are the same type (#6275, @michaelchirico)
```

Comment:
Added the suggestion, but I had in mind to revisit once the full scope of the PR is clearer